HanksMineralEmporium/Service/AuthenticationService/ AuthenticationService.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Web;
+using HanksMineralEmporium.Core.UserManagement;
+using HanksMineralEmporium.Core.UserManagement.Exception;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using MudBlazor;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Http;
+using HanksMineralEmporium.Service.AuthenticationService.Exception;
+/// <summary>
+/// Initializes a new instance of the <see cref="AuthenticationService"/> class.
+/// </summary>
+/// <param name="userManager">An implementation of <see cref="IUserManager"/> for user management.</param>
+/// <param name="serviceProvider">Service provider for dependency injection.</param>
+/// <remarks>
+/// The <paramref name="userManager"/> parameter should not be null.
+/// </remarks>
+/// <summary>
+/// Validates user credentials and attempts to log in the user.
+/// </summary>
+/// <param name="username">The username of the user.</param>
+/// <param name="password">The password of the user.</param>
+
+namespace HanksMineralEmporium.Service.AuthenticationService;
+
+public class  AuthenticationService : IAuthenticationService
+{
+    private IUserManager _userManager;
+    public AuthenticationService(IUserManager userManager)
```

Comment:
You should pass an IHttpContextAccessor into the constructor and make a private field for it. Currently, you are creating a DefaultHttpContext object inside of StoreUserInSession. This means the cookies will not be saved at all since the object immediately falls out of scope and is destroyed by the garbage collector

Reply:
What do you want me to with the IHttpContextAccessor object?